Assets/Scripts/Switch.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Switch : MonoBehaviour
+{
+    [SerializeField]
+    [Tooltip("The color of the switch. 0 if yellow, 1 if blue, 2 if red, 3 if green.")]
+    [Range(0, 3)]
+    private int switchColor;
+
+    [SerializeField]
+    [Tooltip("The puzzle this switch is connected to.\nMust have a script which implements the IPuzzle interface.")]
+    private GameObject connectedPuzzle;
+
+    [SerializeField]
+    [Tooltip("The character this switch adds to the solution.")]
+    private char solutionEntry = '1';
+
+    /// <summary>
+    /// The animator component attached to the switch
+    /// </summary>
+    private Animator animator;
+
+    /// <summary>
+    /// Is the switch active or not?
+    /// </summary>
+    private bool isSwitchOn;
+
+    /// <summary>
+    /// The IPuzzle component of the connected puzzle
```

Comment:
See my comment about the "connected puzzle" gameobject field. I don't think you need both, but if you do, explain.